### PR TITLE
Refactor producer

### PR DESCRIPTION
**Pull request by JiriPapousek**
# Description

Changed producer interface so it receives `[]byte` instead of `types.NotificationMesage`. Also changed file hiearchy.

## Type of change

- Refactor (refactoring code, removing useless files)

## Testing steps

Run locally.

## Checklist
* [x] `make before_commit` passes
* [ ] updated documentation wherever necessary
* [x] added or modified tests if necessary
* [ ] updated schemas and validators in [insights-data-schemas](https://github.com/RedHatInsights/insights-data-schemas) in case of input/output change

### CODE REVIEW

**Review by tisnik — APPROVED** [review]
Nice!

File: producer/disabled/disabled_producer_test.go [hunk 1]
```diff
@@ -14,19 +14,24 @@ See the License for the specific language governing permissions and
...
 	"github.com/RedHatInsights/ccx-notification-service/types"
 	"github.com/stretchr/testify/assert"
 )
 
 func TestDisabledProducer(t *testing.T) {
-	p := producer.DisabledProducer{}
-	_, _, err := p.ProduceMessage(types.NotificationMessage{})
+	p := Producer{}
+
+	msgBytes, err := json.Marshal(types.NotificationMessage{})
```
Comment (juandspy) [inline producer/disabled/disabled_producer_test.go:31]: it can be any type of JSON, right? not necessary of type `types.NotificationMessage{}`

Comment (juandspy) [reply producer/disabled/disabled_producer_test.go:31]: my point is that this can be a bit missleading maybe?

**Review by matysek — APPROVED** [review]

File: producer/disabled/disabled_producer_test.go (hunk 1 above)
Comment (JiriPapousek) [reply producer/disabled/disabled_producer_test.go:31]: In one of the future PRs I might add some checks inside ProduceMessage to verify that the content of `[]bytes` is indeed `types.NotificationMessage{}` :)

Comment (juandspy) [reply producer/disabled/disabled_producer_test.go:31]: will servicelog use `types.NotificationMessage{}` too? in that case, LGTM :)

Comment (JiriPapousek) [reply producer/disabled/disabled_producer_test.go:31]: No, servicelog will actually use different type - I meant to check `types.NotificationMessage`  inside `KafkaProducer`. The fact that `ServiceLogProducer` accepts different data to produce was the reason to replace the type in the first place :)

Comment (juandspy) [reply producer/disabled/disabled_producer_test.go:31]: but the disable producer can accept any kind of data, right? why not use something like
```go
msgBytes, err := json.Marshal([]bytes())
```
?

Comment (JiriPapousek) [reply producer/disabled/disabled_producer_test.go:31]: Oh, I get what you mean now! Ok, I will change it for disable producer then :)

**Review by epapbak — APPROVED** [review]
nice!

**Pull request merged**